src/module_loader.rs

```diff
@@ -89,10 +92,56 @@ impl InnerRustyLoader {
     }
 }
 
+pub trait ImportProvider {

```

Comment:
What was the issue with having a 2nd trait method for resolve?

Reply:
There was no issue with that, I just wanted to wait until after this bit of code is square. I'll add the resolve method while implementing your other suggestions later today.